build-support/bin/generate_travis_yml.py

```diff
@@ -0,0 +1,774 @@
+  }
+
+# -------------------------------------------------------------------------
+# Unit tests
+# -------------------------------------------------------------------------
+
+def unit_tests_v2(python_version: PythonVersion) -> Dict:
+  shard = {
+    **linux_shard(python_version=python_version),
+    "name": f"Unit tests - V2 test runner (Python {python_version.decimal})",
+    "script": [
+      f"travis_wait 50 ./build-support/bin/ci.py --python-tests-v2 --remote-execution-enabled --python-version {python_version.decimal}",
+    ],
+  }
+  shard["env"] = shard.get("env", []) + [f"CACHE_NAME=unit_tests.v2.py{python_version.number}"]
+  return shard
+
+
+def unit_tests_v1(python_version: PythonVersion) -> Dict:
+  shard = {
+    **linux_shard(python_version=python_version),
+    "name": f"Unit tests - V1 test runner (Python {python_version.decimal})",
+    "script": [
+      f"./build-support/bin/ci.py --python-tests-v1 --python-version {python_version.decimal}",
+      f"./build-support/bin/ci.py --plugin-tests --python-version {python_version.decimal}",
+    ],
+  }
+  shard["env"] = shard.get("env", []) + [f"CACHE_NAME=unit_tests.v1.py{python_version.number}"]
+  return shard
+
+# ----------------------------------------------------------------------
+# Build wheels
+# ----------------------------------------------------------------------
+
+def _build_wheels_command() -> List[str]:
+  return [
+    "./build-support/bin/check_pants_pex_abi.py abi3 cp36m",
+    "RUN_PANTS_FROM_PEX=1 ./build-support/bin/release.sh -n",
+  ]
+
+
+def _build_wheels_env(*, platform: Platform) -> List[str]:
+  return [
+    "PREPARE_DEPLOY=1",
+    f"CACHE_NAME=wheels.{platform}.py36",
+  ]
+
+
+def build_wheels_linux() -> Dict:
+  command = " && ".join(_build_wheels_command())
+  shard = {
+    **linux_shard(python_version=PythonVersion.py36, use_docker=True),
+    "name": "Build Linux wheels (Python 3.6)",
+    "script": [
+      docker_build_travis_ci_image(python_version=PythonVersion.py36),
+      docker_run_travis_ci_image(command)
+    ]
+  }
+  shard["env"] = shard.get("env", []) + (_build_wheels_env(platform=Platform.linux))
+  return shard
+
+
+def build_wheels_osx() -> Dict:
+  shard = {
+    **osx_shard(python_version=PythonVersion.py36, osx_image="xcode8"),
+    "name": "Build OSX wheels (Python 3.6)",
+    "script": _build_wheels_command(),
+  }
+  shard["env"] = shard.get("env", []) + _build_wheels_env(platform=Platform.osx) + [
+    # We ensure selection of the pyenv interpreter by PY aware scripts and pants.pex with these
+    # env vars.
+    'PY=${PYENV_ROOT}/versions/${PYENV_PY36_VERSION}/bin/python',
+    """PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="['CPython==${PYENV_PY36_VERSION}']\""""
+  ]
+  return shard
+
+
+# -------------------------------------------------------------------------
+# Integration tests
+# -------------------------------------------------------------------------
+
+NUM_INTEGRATION_SHARDS = 20
```

Comment:
If there isn't a good reason to make this global, maybe just inline it as a local variable in the method. It feels like it could only be error prone for code other than the `integration_tests` method to try and consume this var.

Reply:
Good suggestion! Agreed we should inline.